src/Calculator.php

```diff
@@ -8,44 +8,56 @@
 class Calculator
 {
     private ComposerFile $composer;
-    private PackagistAPI $packagist;
+    private PackageAPI $packageAPI;
 
-    public function __construct(ComposerFile $composer, PackagistAPI $packagist)
+    public function __construct(ComposerFile $composer, PackageAPI $packageAPI)
     {
         $this->composer = $composer;
-        $this->packagist = $packagist;
+        $this->packageAPI = $packageAPI;
     }
 
     /**
      * @param string $directory
      * @return Dependency[]
      */
-    public function getDependencyInfo(string $directory): array
-    {
-        $dependencies = $this->composer->getDependencies($directory);
+	public function getDependencyInfo(string $directory): array
+	{
+		$dependencies = $this->composer->getDependencies($directory);
+		$repositories = array_map(
+			fn($repositoryUrl) => $this->packageAPI->getRepositoryInfo($repositoryUrl),
+			$this->composer->getRepositoriesUrl($directory) ?: ['https://repo.packagist.org']
```

Comment:
Let's move this default URL array inside of `getRepositoryURLs`

Reply:
Done :)